returnn/tf/engine.py

```diff
@@ -1091,6 +1089,7 @@ def init_train_from_config(self, config=None, train_data=None, dev_data=None, ev
             from returnn.datasets.basic import set_config_extern_data_from_dataset
 
             set_config_extern_data_from_dataset(config=config, dataset=train_data or dev_data or eval_data)
+        super().init_train_from_config(config=config)
```

Comment:
This should be as early as possible, i.e. right after `if not config: config = self.config`.